### PR TITLE
Add a column for self visit date in the ART linelist

### PR DESCRIPTION
**Pull request by CynthiaKamau**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/otz/PatientsOnOTZReportBuilder.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
 import org.openmrs.module.kenyacore.report.data.patient.definition.CalculationDataDefinition;
 import org.openmrs.module.kenyaemr.calculation.library.hiv.art.CurrentArtRegimenCalculation;
 import org.openmrs.module.kenyaemr.calculation.library.hiv.art.ViralLoadResultCalculation;
+import org.openmrs.module.kenyaemr.reporting.data.converter.definition.art.ETLLastVisitDateDataDefinition;
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/otz/PatientsOnOTZReportBuilder.java:21]: Please move these definitions and their evaluators to the otz package. That will help distinguish them from other similar definitions

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/ETLHivSelfVisitDateDataEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@Handler(supports=ETLHivSelfVisitDateDataDefinition.class, order=50)
+public class ETLHivSelfVisitDateDataEvaluator implements PersonDataEvaluator {
+
+    @Autowired
+    private EvaluationService evaluationService;
+
+    public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
+        EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
+
+        String qry="select patient_id, max(visit_date) as last_visit_date from kenyaemr_etl.etl_patient_hiv_followup where person_present=978 GROUP BY patient_id;";
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/ETLHivSelfVisitDateDataEvaluator.java:40]: looks good

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/art/ETLHivSelfVisitDateDataDefinition.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        super(name);
+    }
+
+    //***** INSTANCE METHODS *****
+
+    /**
+     * @see org.openmrs.module.reporting.data.DataDefinition#getDataType()
+     */
+    public Class<?> getDataType() {
+        return Double.class;
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/art/ETLHivSelfVisitDateDataDefinition.java:45]: This should be Date.class

**Review by ojwanganto — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/otz/PatientsOnOTZReportBuilder.java (hunk 1 above)
Comment (CynthiaKamau) [reply api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/otz/PatientsOnOTZReportBuilder.java:21]: This one is for the art linelist, not otz

**Review by makombe — COMMENTED** [review]
LGTM

**Review by makombe — APPROVED** [review]

**Pull request merged**